### PR TITLE
staging/publishing: add release-1.21 branch

### PR DESCRIPTION
**Pull request by nikhita**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Needed to publish v1.21 tags

#### Special notes for your reviewer:

See [slack thread](https://kubernetes.slack.com/archives/C2C40FMNF/p1617007926111900) for more context

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

/assign @xmudrii 
cc @soltysh @palnabarun

### CODE REVIEW

**Review by soltysh — APPROVED** [review]
/lgtm

File: staging/publishing/rules.yaml [hunk 1]
```diff
@@ -34,6 +34,11 @@ rules:
       branch: release-1.20
       dir: staging/src/k8s.io/code-generator
     name: release-1.20
+    go: 1.15.10
```
Comment (AlekSi) [inline staging/publishing/rules.yaml:37]: I wonder if that should be 1.16.1 after https://github.com/kubernetes/kubernetes/pull/100106

Comment (soltysh) [reply staging/publishing/rules.yaml:37]: no, this is for 1.20, which should use go1.15, only 1.21 and master are on 1.16

Comment (nikhita) [reply staging/publishing/rules.yaml:37]: @AlekSi that's for the master branch, release-1.20 uses go1.15.10 :)

https://github.com/kubernetes/kubernetes/blob/release-1.20/build/build-image/cross/VERSION

Comment (AlekSi) [reply staging/publishing/rules.yaml:37]: Doh, missed that `-` is below. Thank you!

**Review by AlekSi — APPROVED** [review]
:shipit:

**Review by xmudrii — APPROVED** [review]
/lgtm

**Pull request merged**